test/governance/extensions/GovernorTimelockAccess.test.js

```diff
@@ -141,6 +223,107 @@ contract('GovernorTimelockAccess', function (accounts) {
+        await this.helper.vote({ support: Enums.VoteType.For }, { from: voter1 });
+        await this.helper.waitForDeadline();
+        await this.helper.queue();
+        await this.manager.grantRole(roleId, this.mock.address, 0, { from: admin }); // Purposefully remove delay so the returned nonce differs
+        await this.helper.waitForEta();
+        await expectRevertCustomError(this.helper.execute(), 'GovernorMismatchedNonce', [this.proposal.id, 1, 0]);
+      });
+
+      it('reverts with a proposal including multiple operations but one of those was cancelled in the manager', async function () {
+        const delay = time.duration.hours(2);
+        const roleId = '1';
+
+        await this.manager.setTargetFunctionRole(this.receiver.address, [this.restricted.selector], roleId, {
+          from: admin,
+        });
+        await this.manager.grantRole(roleId, this.mock.address, delay, { from: admin });
+
+        // Set original proposal
+        this.proposal = await this.helper.setProposal(
+          [this.restricted.operation, this.unrestricted.operation],
+          'descr',
+        );
+
+        // Go through all the governance process
+        await this.helper.propose();
+        expect(await this.mock.proposalNeedsQueuing(this.proposal.id)).to.be.eq(true);
+        const { delay: planDelay, indirect, withDelay } = await this.mock.proposalExecutionPlan(this.proposal.id);
+        expect(planDelay).to.be.bignumber.eq(web3.utils.toBN(delay));
+        expect(indirect).to.deep.eq([true, false]);
+        expect(withDelay).to.deep.eq([true, false]);
+        await this.helper.waitForSnapshot();
+        await this.helper.vote({ support: Enums.VoteType.For }, { from: voter1 });
+        await this.helper.waitForDeadline();
+        await this.helper.queue();
+        await this.helper.waitForEta();
+
+        // Suddenly cancel one of the proposed operations in the manager
+        await this.manager.cancel(this.mock.address, this.restricted.operation.target, this.restricted.operation.data, {
+          from: admin,
+        });
+
+        // Reschedule the same operation in a different proposal to avoid "AccessManagerNotScheduled" error
+        this.proposal = await this.helper.setProposal([this.restricted.operation], 'descr');
+        await this.helper.propose();
+        await this.helper.waitForSnapshot();
+        await this.helper.vote({ support: Enums.VoteType.For }, { from: voter1 });
+        await this.helper.waitForDeadline();
+        await this.helper.queue(); // This will schedule it again in the manager
+        await this.helper.waitForEta();
+
+        // Set original proposal back for executing
+        this.proposal = await this.helper.setProposal(
+          [this.restricted.operation, this.unrestricted.operation],
+          'descr',
+        );
+
+        // Attempt to execute
+        await expectRevertCustomError(this.helper.execute(), 'GovernorMismatchedNonce', [this.proposal.id, 1, 2]);
```

Comment:
```suggestion
        // Set proposals
        const proposal1 = new GovernorHelper(this.mock, mode);
        await helper1.setProposal([this.restricted.operation, this.unrestricted.operation], 'descr');

        const proposal2 = new GovernorHelper(this.mock, mode);
        await this.helper.setProposal([this.restricted.operation], 'descr');

        // Go through all the governance process
        await proposal1.propose();
        expect(await this.mock.proposalNeedsQueuing(proposal1.currentProposal.id)).to.be.eq(true);
        const { delay: planDelay, indirect, withDelay } = await this.mock.proposalExecutionPlan(proposal1.currentProposal.id);
        expect(planDelay).to.be.bignumber.eq(web3.utils.toBN(delay));
        expect(indirect).to.deep.eq([true, false]);
        expect(withDelay).to.deep.eq([true, false]);
        await proposal1.waitForSnapshot();
        await proposal1.vote({ support: Enums.VoteType.For }, { from: voter1 });
        await proposal1.waitForDeadline();
        await proposal1.queue();
        await proposal1.waitForEta();

        // Suddenly cancel one of the proposed operations in the manager
        await this.manager.cancel(this.mock.address, this.restricted.operation.target, this.restricted.operation.data, {
          from: admin,
        });

        // Reschedule the same operation in a different proposal to avoid "AccessManagerNotScheduled" error
        await proposal2.propose();
        await proposal2.waitForSnapshot();
        await proposal2.vote({ support: Enums.VoteType.For }, { from: voter1 });
        await proposal2.waitForDeadline();
        await proposal2.queue(); // This will schedule it again in the manager
        await proposal2.waitForEta();

        // Attempt to execute
        await expectRevertCustomError(proposal1.execute(), 'GovernorMismatchedNonce', [proposal1.currentProposal.id, 1, 2]);
```